zjit/src/codegen.rs

```diff
@@ -420,6 +420,7 @@ fn gen_insn(cb: &mut CodeBlock, jit: &mut JITState, asm: &mut Assembler, functio
         Insn::GetSpecialSymbol { symbol_type, state: _ } => gen_getspecial_symbol(asm, *symbol_type),
         Insn::GetSpecialNumber { nth, state } => gen_getspecial_number(asm, *nth, &function.frame_state(*state)),
         &Insn::IncrCounter(counter) => no_output!(gen_incr_counter(asm, counter)),
+        Insn::IncrDynamicCounter { counter_ptr } => no_output!(gen_incr_counter_ptr(asm, *counter_ptr)),
```

Comment:
> https://github.com/ruby/ruby/pull/14638#discussion_r2383005725

I got rather confused by the "Dynamic" part of the name. I get that `counter_ptr` is dynamically fetched, but once the instruction is constructed, it seems no more dynamic than `IncrCounter`.

How about calling it `IncrCounterPtr`?